### PR TITLE
dimension measure panel divider resize

### LINKED ISSUE #48: Ability to resize dimensions and measures panels
For data cubes with long list of dimensions and short list of measures (or long list of measures and short list of dimensions) the screen viewport is not fully utilized. The panel boundary between dimensions and measures panels should be draggable.

### LINKED ISSUE #49: Ability to hide side panels and make main visualization panel full screen.
On smaller screens I would like hide side panels, and make main visualization bigger. 
This Hide/Show feature could be also applied to filter/split area.

### PR DESCRIPTION
**Pull request by adrianmroz**
Closes #48 
Closes #49

### CODE REVIEW

File: src/client/views/cube-view/cube-view.scss [hunk 1]
```diff
@@ -47,28 +48,56 @@
...
+
+      .dimension-panel-toggle, .pinboard-toggle {
+        display: flex;
+        align-items: center;
+        height: 100%;
+        background-color: $background-lightest;
+        cursor: pointer;
+
+        svg {
+          width: $toggle-width - 3;
```
Comment (pguzik) [inline src/client/views/cube-view/cube-view.scss]: magic number - do not know why 3 ?

File: src/client/views/cube-view/cube-view.tsx [hunk 2]
```diff
@@ -407,27 +428,60 @@ export class CubeView extends React.Component<CubeViewProps, CubeViewState> {
...
+      dimensionPanel: {
+        ...dimensionPanel,
+        hidden: !dimensionPanel.hidden
+      }
+    });
+  }
 
-    this.setState({ layout });
-    this.storeLayout(layout);
+  togglePinboard() {
```
Comment (pguzik) [inline src/client/views/cube-view/cube-view.tsx]: a bit of a duplication down here 3 times :)

Comment (adrianmroz) [reply src/client/views/cube-view/cube-view.tsx]: Even four times. I need some solution for deep updates in objects but didn't have courage to pick new dependency for that.

File: src/client/views/cube-view/cube-view.scss (hunk 1 above)
Comment (adrianmroz) [reply src/client/views/cube-view/cube-view.scss]: I eyeballed it. I think I could leave `3` but should name this expression as `$toggle-icon-width`

File: src/client/views/cube-view/cube-view.tsx [hunk 3]
```diff
@@ -407,27 +428,60 @@ export class CubeView extends React.Component<CubeViewProps, CubeViewState> {
   }
 
   getStoredLayout(): CubeViewLayout {
-    return localStorage.get("cube-view-layout") || { dimensionPanelWidth: 240, pinboardWidth: 240 };
+    return localStorage.get("cube-view-layout-ver2") || defaultLayout;
```
Comment (mkuthan) [inline src/client/views/cube-view/cube-view.tsx]: v2 seems to be enough

File: src/client/views/cube-view/cube-view.tsx [hunk 4]
```diff
@@ -59,11 +60,27 @@ import { getVisualizationComponent } from "../../visualizations/index";
...
+    hidden?: boolean;
+  };
+  pinboard: {
+    width: number;
+    hidden?: boolean;
+  };
 }
 
+const defaultLayout: CubeViewLayout = {
+  dimensionPanel: { width: 240 },
```
Comment (mkuthan) [inline src/client/views/cube-view/cube-view.tsx]: This is rather factPanel (dims + measurements)

File: src/client/views/cube-view/cube-view.tsx (hunk 2 above)
Comment (adrianmroz) [reply src/client/views/cube-view/cube-view.tsx]: Will be resolved in #202

**Pull request merged**